server/aws-lsp-identity/src/language-server/utils.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    Chat,
+    CredentialsProvider,
+    IdentityManagement,
+    Logging,
+    Lsp,
+    Notification,
+    Runtime,
+    Workspace,
+    PartialInitializeResult,
+    Telemetry,
+} from '@aws/language-server-runtimes/server-interface'
+import { InitializeParams, HandlerResult, InitializeError } from 'vscode-languageserver'
+
+export function ensureSsoAccountAccessScope(scopes?: string[]): string[] {
+    const ssoAccountAccessScope = 'sso:account:access'
+
+    if (!scopes) {
+        scopes = [ssoAccountAccessScope]
+    } else if (!scopes.includes(ssoAccountAccessScope)) {
+        scopes.push(ssoAccountAccessScope)
+    }
+
+    return scopes
+}
+
+export interface ServerFeatures {
+    chat: Chat
+    credentialsProvider: CredentialsProvider
+    identityManagement: IdentityManagement
+    logging: Logging
+    lsp: Lsp
+    notification: Notification
+    runtime: Runtime
+    telemetry: Telemetry
+    workspace: Workspace
+}
+
+export interface Observability {
+    logging: Logging
+    telemetry: Telemetry
+}
+
+export abstract class ServerBase implements Disposable {
+    protected readonly observability: Observability
+    protected readonly disposables: (Disposable | (() => void))[] = []
+
+    constructor(protected readonly features: ServerFeatures) {
+        this.features.lsp.addInitializer(params => this.initialize(params))
+        this.observability = { logging: this.features.logging, telemetry: this.features.telemetry }
```

Comment:
nit: Having `features` and `observability` as separate fields on the level of server does not look super clean to me - observability is just a subset of features, both objects give access to the same stuff.

Reply:
I agree its redundant, but the Observability interface was more for having to thread both logging and telemetry through the constructors of almost every class/function in the aws-lsp-identity.  I was trying to reduce the addition of multiple parameters everywhere and not have to thread new observability features through all of the code again (i.e. just add them to the Observability interface and they become available everywhere).  

I could remove the ServerBase.observability property, but I'd still have to create a instance during the initialize call to pass around to all of the dependents.  For reuse, it seemed easer to just have a property containing an instance ready for use at any time.  Every subclass will need to use it in its initialize method unless it doesn't emit observability data for some reason.